### ISSUE TITLE
Enable choice of SSH port number in the bootstrap page for Salt SSH minions

### ISSUE THREAD

**Issue by paususe** [uyuni-project/uyuni#3412]
There are many cases where people might want SSH to run on unusual ports:

* in https://bugzilla.suse.com/show_bug.cgi?id=1175849, I describe a method to traverse NAT (issue #11839) using a SSH jump host listening on port 2222;
* sometimes it's needed to traverse firewalls on ports different than port 22;
* some admins do experiments on port 23 before switching to regular port 22;
* whatever other reason the user might have.

We can bootstrap minions using other ports than port 22 for normal minions. However, the "SSH port" text box is greyed out in the Salt SSH case.

Definition of done:
* [ ] Add support for ports other than 22 in the database and the code
* [ ] Do not grey out the port number field in the web UI

**Comment by agraul** [comment]
@paususe, @Bischoff  can we open the bug report so Uyuni contributors can read it as well?

**Comment by paususe** [comment]
> @paususe, @Bischoff can we open the bug report so Uyuni contributors can read it as well?

I don't think I have superpowers enough but I can copy the procedure @Bischoff was talking about here:


How to reproduce
================

Go to Systems => Bootstrapping.

Tick the "Manage system completely via SSH (will not install an agent)" box.


What happens
============

The "SSH port" text entry box is greyed out.


What should happen
==================

This box should not be greyed out. It should be possible to select the port on which SSH runs, just like in the "normal minion" case.

I don't see what makes Salt SSH different from Salt in terms of SSH port numbers.


Why this would be needed
========================

There might be several use cases, but here is one:

If your minion is running behing NAT, you might want to access it via a jump host that has a public IP, unlike the minion.

You would reverse tunnel the minion's SSH port 22 onto the jump host, on some other port, for example 2222:

  # ssh <jump_host_public_address> -R '*:2222:<minion_private_address>:22'

From Uyuni, you want to connect to jump host's 2222 port via Salt SSH.

**Comment by wombelix** [comment]
Hey @agraul,
I'm looking for a good-first-issue and want to check if this one still has some relevance.

Also the described use case sounds more like using a SSH Jumphost, even after the initial bootstrap, instead just an individual SSH Port during the bootstrap Phase?
So my understanding is, that this could already be achieved, by just setting up a Uyuni Proxy?

> @paususe, @Bischoff can we open the bug report so Uyuni contributors can read it as well?

Unfortunately I see quite a lot links which are not public visible (SUSE Bugzilla, Confluence, Github Repos...), makes it sometimes hard to follow a discussion, the Idea behind a request or if it's still up to date :(

Cheers
Dominik

**Comment by mcalmer** [comment]
The issue is still not done, but I am not sure if this is really a "good first issue" and from my point of view it requires a lot of knowledge.

Problem: we talk about permanent salt-ssh managed clients here. This means we need to store the ssh port in the DB for this client. This requires a DB schema change together with java hibernate changes, etc. 

We also have already global configuration option for the ssh port. The requirement is, that all ssh daemons which want to be used for salt-ssh management, use the same port. This might be true or not.

And with jump hosts having again different ports the problem gets even more complex.

**Comment by juliogonzalez** [comment]
Closing. Was implemented in December last year.
